src/compiler/utilitiesPublic.ts

```diff
@@ -2596,31 +2594,8 @@ export function isRestParameter(node: ParameterDeclaration | JSDocParameterTag):
     return (node as ParameterDeclaration).dotDotDotToken !== undefined || !!type && type.kind === SyntaxKind.JSDocVariadicType;
 }
 
-function hasInternalAnnotation(range: CommentRange, sourceFile: SourceFile) {
-    const comment = sourceFile.text.substring(range.pos, range.end);
-    return comment.includes("@internal");
-}
-
 export function isInternalDeclaration(node: Node, sourceFile?: SourceFile) {
     sourceFile ??= getSourceFileOfNode(node);
     const parseTreeNode = getParseTreeNode(node);
-    if (parseTreeNode && parseTreeNode.kind === SyntaxKind.Parameter) {
-        const paramIdx = (parseTreeNode.parent as SignatureDeclaration).parameters.indexOf(parseTreeNode as ParameterDeclaration);
-        const previousSibling = paramIdx > 0 ? (parseTreeNode.parent as SignatureDeclaration).parameters[paramIdx - 1] : undefined;
-        const text = sourceFile.text;
-        const commentRanges = previousSibling
-            ? concatenate(
-                // to handle
-                // ... parameters, /** @internal */
-                // public param: string
-                getTrailingCommentRanges(text, skipTrivia(text, previousSibling.end + 1, /*stopAfterLineBreak*/ false, /*stopAtComments*/ true)),
-                getLeadingCommentRanges(text, node.pos),
-            )
-            : getTrailingCommentRanges(text, skipTrivia(text, node.pos, /*stopAfterLineBreak*/ false, /*stopAtComments*/ true));
-        return some(commentRanges) && hasInternalAnnotation(last(commentRanges), sourceFile);
-    }
-    const leadingCommentRanges = parseTreeNode && getLeadingCommentRangesOfNode(parseTreeNode, sourceFile);
-    return !!forEach(leadingCommentRanges, range => {
-        return hasInternalAnnotation(range, sourceFile);
-    });
+    return parseTreeNode && !!getJSDocInternalTag(parseTreeNode);

```

Comment:
```suggestion
    return !!parseTreeNode && !!getJSDocInternalTag(parseTreeNode);
```

This function is public and should continue to return just boolean.